### PR TITLE
Introduce new TIL string helpers, teach older ones to use wmemcmp

### PR DESCRIPTION
**Pull request by lhecker**
til::equals:
At the time of writing wmemcmp() is not an intrinsic for MSVC,
but the STL uses it to implement wide string comparisons.
This produces 3x the assembly _per_ comparison and increases
runtime by 2-3x for strings of medium length (16 characters)
and 5x or more for long strings (128 characters or more).
See: https://github.com/microsoft/STL/issues/2289

Additionally a number of case insensitive, locale unaware
helpers for prefix/suffix comparisons are introduced.

### CODE REVIEW

**Comment by lhecker** [comment]
Please let me know what you think about this. 🙂

**Review by DHowett — DISMISSED** [review]
Seems fine to me!

**Comment by DHowett** [comment]
@msftbot make sure @miniksa signs off

**Comment by DHowett** [comment]
Make sure (1) it actually builds and (2) very specifically, that the full solution builds on ARM64

**Comment by lhecker** [comment]
I can't get this to compile properly so I'll pivot and add a different method (`til::equal`) for future use...

**Comment by lhecker** [comment]
So yeah I've reverted almost all of the changes now because I can't for my life figure out how to overload WEX's `VerifyEqual` struct to use my overloads and not run in ambiguity errors:
```
error C2593: 'operator ==' is ambiguous (compiling source file ConptyOutputTests.cpp)
could be 'bool operator ==(const std::wstring &,const std::wstring_view &) noexcept' (compiling source file ConptyOutputTests.cpp)
or       'bool operator ==(const std::wstring_view &,const std::wstring_view &) noexcept' (compiling source file ConptyOutputTests.cpp)
or       'bool std::operator ==<wchar_t,std::char_traits<wchar_t>,1>(const std::basic_string_view<wchar_t,std::char_traits<wchar_t>>,const std::basic_string_view<wchar_t,std::char_traits<wchar_t>>) noexcept' (compiling source file ConptyOutputTests.cpp)
while trying to match the argument list '(const wchar_t [2], const T2)'
    with
    [
        T2=std::wstring_view
    ] (compiling source file ConptyOutputTests.cpp)
```

**Review by miniksa — COMMENTED** [review]
OK

File: src/inc/operators.hpp [hunk 1]
```diff
@@ -37,23 +37,3 @@ constexpr bool operator!=(const SMALL_RECT& a, const SMALL_RECT& b) noexcept
 {
     return !(a == b);
 }
-
-constexpr bool operator==(const std::wstring& wstr, const std::wstring_view& wstrView)
-{
-    return (wstrView == std::wstring_view{ wstr.c_str(), wstr.size() });
-}
```
Comment (DHowett) [inline src/inc/operators.hpp:44]: why's it still safe to get rid of these? are they just builtin now as of C++17 or so?

Comment (lhecker) [reply src/inc/operators.hpp:44]: I'm fairly certain about that. Otherwise it wouldn’t compile…

Comment (lhecker) [reply src/inc/operators.hpp:44]: Yep, a comparison of `std::wstring_view` with `std::wstring_view` compiles to the same assembly as `std::wstring` and `std::wstring_view`.

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
derp i renamed it to "to use wmemcmp" when i meant "to use memcmp". Oh well.

**Pull request merged**